runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkPipelineExecutionEnvironment.java

```diff
@@ -95,73 +83,33 @@ public void translate(Pipeline pipeline) {
-    } else {
-      filesToStage = options.getFilesToStage();
-    }
+    prepareFilesToStageForRemoteClusterExecution();
 
     FlinkPipelineTranslator translator;
     if (translationMode == TranslationMode.STREAMING) {
       this.flinkStreamEnv =
-          FlinkExecutionEnvironments.createStreamExecutionEnvironment(options, filesToStage);
+          FlinkExecutionEnvironments.createStreamExecutionEnvironment(
+              options, options.getFilesToStage());
       translator = new FlinkStreamingPipelineTranslator(flinkStreamEnv, options);
     } else {
       this.flinkBatchEnv =
-          FlinkExecutionEnvironments.createBatchExecutionEnvironment(options, filesToStage);
+          FlinkExecutionEnvironments.createBatchExecutionEnvironment(
+              options, options.getFilesToStage());
       translator = new FlinkBatchPipelineTranslator(flinkBatchEnv, options);
     }
 
     translator.translate(pipeline);
   }
 
-  private List<String> prepareFilesToStage() {
-    return options
-        .getFilesToStage()
-        .stream()
-        .map(File::new)
-        .filter(File::exists)
-        .map(file -> file.isDirectory() ? packageDirectoriesToStage(file) : file.getAbsolutePath())
-        .collect(Collectors.toList());
-  }
-
-  private String packageDirectoriesToStage(File directoryToStage) {
-    String hash = calculateDirectoryContentHash(directoryToStage);
-    String pathForJar = getUniqueJarPath(hash);
-    zipDirectory(directoryToStage, pathForJar);
-    return pathForJar;
-  }
-
-  private String calculateDirectoryContentHash(File directoryToStage) {
-    Hasher hasher = Hashing.md5().newHasher();
-    try (OutputStream hashStream = Funnels.asOutputStream(hasher)) {
-      ZipFiles.zipDirectory(directoryToStage, hashStream);
-      return Base64Variants.MODIFIED_FOR_URL.encode(hasher.hash().asBytes());
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-  }
-
-  private String getUniqueJarPath(String contentHash) {
-    String tempLocation = options.getTempLocation();
-
-    checkArgument(
-        !Strings.isNullOrEmpty(tempLocation),
-        "Please provide \"tempLocation\" pipeline option. Flink runner needs it to store jars "
-            + "made of directories that were on classpath.");
-
-    return String.format("%s%s.jar", tempLocation, contentHash);
-  }
-
-  private void zipDirectory(File directoryToStage, String uniqueDirectoryPath) {
-    try {
-      ZipFiles.zipDirectory(directoryToStage, new FileOutputStream(uniqueDirectoryPath));
-    } catch (IOException e) {
-      throw new RuntimeException(e);
+  /**
+   * Local configurations work in the same JVM and have no problems with improperly formatted files
+   * on classpath (eg. directories with .class files or empty directories). Prepare files for
+   * staging only when using remote cluster.
+   */
+  private void prepareFilesToStageForRemoteClusterExecution() {
+    if (!options.getFlinkMaster().matches("\\[.*\\]")) {
```

Comment:
Can we just check that `getFlinkMaster()` is not empty?

Reply:
So the idea here is that we don't want to `prepareFilesForStaging()` when using local (and auto, which in turn can decide to run [local]). We only need to prepare resources like this (remove nonexistent paths, package test resources into jars etc) when we actually stage them to a remote runner. 

I figured that it might be more descriptive when the code will be more explicit and mention "[auto]" and "[local]" here, wdyt?

Reply:
I agree with `[local]` but not sure about `[auto]` since it depends on the environment and, in the end, it could be run on remote cluster. In this case, staging won't be applied, right?

Reply:
This is unfortunately true. I must admit that this is a kind of a bitter-sweet compromise to me but couldn't find some other sensible option. 

I didn't find a better way to recognize if we're running on a local or remote cluster. I don't think that this deserves eg. a separate pipeline option (`eg --integrationTest`) due to two reasons: 
 - this is a production runner, not a test one so I wouldn't add such test-related flags to it because it is a "noise" for runner users
 - we are not sure if this mechanism won't be useful in some other cases

I decided to do it like this at least for now, leaving a descriptive comment for the future. We can always develop it more. Do you think this is enough?

BTW: I was missing [collection] here - it also shouldn't be used. I'll post the fix.

Reply:
I'm ok with that but I'd double check if this doesn't break something. @mxm, wdyt?